gatling-http/src/main/scala/io/gatling/http/action/ws/WsFrameCheckSequenceBuilder.scala

```diff
@@ -25,13 +26,43 @@ import io.gatling.http.check.ws.{ WsFrameCheck, WsFrameCheckSequence }
 object WsFrameCheckSequenceBuilder {
 
   @SuppressWarnings(Array("org.wartremover.warts.ListAppend"))
-  def resolve[T <: WsFrameCheck](builders: List[WsFrameCheckSequenceBuilder[T]], session: Session): Validation[List[WsFrameCheckSequence[T]]] =
+  def resolve[T <: WsFrameCheck](builders: List[WsFrameCheckSequenceBuilder[T]], session: Session): Validation[List[WsFrameCheckSequence[T]]] = {
     builders.foldLeft(List.empty[WsFrameCheckSequence[T]].success) { (acc, builder) =>
       for {
-        accValue <- acc
         timeout <- builder.timeout(session)
-      } yield accValue :+ WsFrameCheckSequence(timeout, builder.checks)
+        accValue <- acc
+        checks <- resolveChecks(builder.checks, session)
+      } yield accValue :+ WsFrameCheckSequence(timeout, checks)
+    }
+  }
+
+  @SuppressWarnings(Array("org.wartremover.warts.ListAppend"))
+  private def resolveChecks[T <: WsFrameCheck](checks: List[T], session: Session): Validation[List[T]] = {
+
+    @tailrec
+    def resolveCheckName[C <: WsFrameCheck](entries: Iterator[C], session: Session, acc: List[C]): Validation[List[C]] = {
```

Comment:
You mustn't use an Iterator for that, that's not the Scala way.
You must pattern match the List, see https://github.com/gatling/gatling/blob/main/gatling-core/src/main/scala/io/gatling/core/check/Check.scala#L47-L61 for an example